vllm/core/scheduler.py

```diff
@@ -1364,9 +1449,30 @@ def schedule(
                         < seqs[0].data.get_len()):
                     do_sample = False
 
+            is_remote_prefill = False
+            if is_first_prefill and seq_group.remote_prefill_params is not None and seq_group.remote_prefill_params.is_remote_prefill:
+                is_remote_prefill = True
+                logger.debug("Remote prefill, computed block nums: %s", common_computed_block_nums)
+            if is_first_prefill and seq_group.remote_prefill_params is not None and seq_group.remote_prefill_params.is_remote_decode:
+                block_tables[seq_group.seqs[0].seq_id + 1] = self.block_manager.block_tables[seq.seq_id + 1].physical_block_ids
```

Comment:
@kkranen - question for you.

In this case, we are on a `PrefillWorker` (`is_remote_decode=True`)

`seq_id+1` is the sequence id of the staging sequence.

I do not quite follow why we need to add this to the `block_tables` for the `SequenceGroupMetadata` of the actual sequence. Since we call `rearrange_tensors` in `DynamoNixlConnector.write_blocks()`, we have a separate section of the code where we handle actually writing to the staging blocks

Just want to make sure I understand correctly since the staging piece is a bit subtle.

Reply:
Also cc @pavanimajety